### PR TITLE
Add wrappers around PyFrame and PyCode

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

**Comment by robert3005** [comment]
Not sure if this benefits from a docs mention

File: pydust/src/types/code.zig [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+const std = @import("std");
+const py = @import("../pydust.zig");
+const PyObjectMixin = @import("./obj.zig").PyObjectMixin;
+
+const ffi = py.ffi;
+
+/// Wrapper for Python PyCode.
+/// See: https://docs.python.org/3/c-api/code.html
+pub const PyCode = extern struct {
```
Comment (robert3005) [inline pydust/src/types/code.zig:21]: you can find a handy reference of the field names in https://docs.python.org/3/library/inspect.html

File: pydust/src/types/frame.zig [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+const ffi = py.ffi;
+
+/// Wrapper for Python PyFrame.
+/// See: https://docs.python.org/3/c-api/frame.html
+pub const PyFrame = extern struct {
+    obj: py.PyObject,
+
+    pub fn get() ?PyFrame {
+        const frame = ffi.PyEval_GetFrame();
```
Comment (robert3005) [inline pydust/src/types/frame.zig:25]: you can also get a frame from threadstate but that's extension for later

File: pydust/src/types/frame.zig [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    pub fn code(self: PyFrame) py.PyCode {
+        const codeObj = ffi.PyFrame_GetCode(framePtr(self.obj.py));
+        return .{ .obj = .{ .py = @alignCast(@ptrCast(codeObj)) } };
+    }
+
+    pub fn lineNumber(self: PyFrame) u32 {
+        return @intCast(ffi.PyFrame_GetLineNumber(framePtr(self.obj.py)));
+    }
+
+    fn framePtr(obj: *ffi.PyObject) *ffi.PyFrameObject {
```
Comment (gatesn) [inline pydust/src/types/frame.zig]: Either literally inline, or mark the function inline

**Review by gatesn — APPROVED** [review]

**Pull request merged**